react/components/LayoutModeSwitcher.js

```diff
@@ -6,40 +6,32 @@ import Grid from '../images/Grid'
 import SingleItemGrid from '../images/SingleItemGrid'
 import InlineGrid from '../images/InlineGrid'
 
-const onChangeHandler = (type, onChange) => e => onChange(e, type)
 
 export default function LayoutModeSwitcher({ activeMode, onChange }) {
-  return (
-    <div className="vtex-layout-switcher flex justify-between pv2">
-      <div className="flex justify-center flex-auto br b--muted-4">
-        <Button
-          variation="tertiary"
-          size="small"
-          onClick={onChangeHandler('small', onChange)}
-        >
-          <Grid active={activeMode === 'small'} />
-        </Button>
-      </div>
-      <div className="flex justify-center flex-auto br b--muted-4">
-        <Button
-          variation="tertiary"
-          size="small"
-          onClick={onChangeHandler('inline', onChange)}
-        >
-          <InlineGrid active={activeMode === 'inline'} />
-        </Button>
-      </div>
-      <div className="flex justify-center flex-auto">
-        <Button
-          variation="tertiary"
-          size="small"
-          onClick={onChangeHandler('normal', onChange)}
-        >
-          <SingleItemGrid active={activeMode === 'normal'} />
-        </Button>
+
+  const onChangeHandler = (type, onChange) => e => onChange(e, type)
+  const renderButton = (mode, onChange) => {
+
+    const renderIcon = mode => {
+      return mode === 'small' ? <Grid /> : mode === 'inline' ? <InlineGrid /> : <SingleItemGrid />
```

Comment:
I think that a switch would be more easily readable than this ternary. What do you think?

Reply:
I Agree with you. I prefered to use if-else statement.